### PR TITLE
add spec.paused field to pause the syncing of tidb cluster

### LINKED ISSUE #1381: support pausing rolling-update of TidbCluster
## Feature Request

Scenarios:

- For critical env, `pause` could be used to avoid unexpected rolling-update caused by a compatible issue of new operator version or an operation error (user fault).
- Support user to modify the spec multiple times and only apply once, or only apply at maintenance window.

Reference:

- Pause support of Deployment

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 

fixes #1381

### What is changed and how does it work?

when the spec.paused is set to true, the following components are paused:

- pd service and statefulset (including configmap)
- tikv service and statefulset (including configmap)
- tidb service and statefulset (including configmap)
- pump service and statefulset (including configmap)
- discovery deployment (not sure yet)
- tidb initializer (not sure yet)

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Able to pause and unpause tidb cluster deployment via `spec.paused`.
```

### CODE REVIEW

File: pkg/controller/tidbcluster/tidb_cluster_control.go [hunk 1]
```diff
@@ -114,6 +114,13 @@ func (tcc *defaultTidbClusterControl) validate(tc *v1alpha1.TidbCluster) bool {
 }
 
 func (tcc *defaultTidbClusterControl) updateTidbCluster(tc *v1alpha1.TidbCluster) error {
+	// Skip syncing if the tidb cluster is paused.
+	if tc.Spec.Paused {
+		// TODO when we record condition of the tidb cluster, we can emit event on state change
```
Comment (cofyc) [inline pkg/controller/tidbcluster/tidb_cluster_control.go]: https://github.com/pingcap/tidb-operator/issues/2011

**Comment by Yisaer** [comment]
Do we need to add release note?

**Comment by cofyc** [comment]
> Do we need to add release note?

yes, I'll add.

**Comment by DanielZhangQD** [comment]
@cofyc Please also update docs-tidb-operator for this feature.

**Comment by cofyc** [comment]
> @cofyc Please also update docs-tidb-operator for this feature.

a track issue is created: https://github.com/pingcap/tidb-operator/pull/2013#issuecomment-604994927

**Pull request merged**